File: src/main/java/icu/samnyan/aqua/sega/chusan/model/userdata/UserActivity.java
package icu.samnyan.aqua.sega.chusan.model.userdata;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author samnyan (privateamusement@protonmail.com)
 */
@Entity(name = "ChusanUserActivity")
@Table(name = "chusan_user_activity", uniqueConstraints = {@UniqueConstraint(columnNames = {"user_id", "kind", "activity_id"})})
@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonPropertyOrder({"kind", "id", "sortNumber", "param1", "param2", "param3", "param4"})
public class UserActivity extends Chu3UserEntity {
    private int kind;

    @JsonProperty("id")
    @Column(name = "activity_id")
    private int activityId;

    private int sortNumber;

    private int param1;

    private int param2;

    private int param3;

    private int param4;

    public UserActivity(Chu3UserData userData) { setUser(userData); }
}

File: src/main/java/icu/samnyan/aqua/sega/chusan/model/userdata/UserActivity.kt
package icu.samnyan.aqua.sega.chusan.model.userdata

import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Table
import jakarta.persistence.UniqueConstraint

@Entity(name = "ChusanUserActivity")
@Table(name = "chusan_user_activity", uniqueConstraints = [UniqueConstraint(columnNames = ["user_id", "kind", "activity_id"])])
class UserActivity(user: Chu3UserData) : Chu3UserEntity() {
    init { this.user = user }

    var kind = 0
    @JsonProperty("id")
    @Column(name = "activity_id")
    var activityId = 0
    var sortNumber = 0
    var param1 = 0
    var param2 = 0
    var param3 = 0
    var param4 = 0
}
